pkg/controller/node/rate_limited_queue.go

```diff
@@ -216,3 +219,30 @@ func (q *RateLimitedTimedQueue) Remove(value string) bool {
 func (q *RateLimitedTimedQueue) Clear() {
 	q.queue.Clear()
 }
+
+// SwapLimiter safely swaps current limiter for this queue with the passed one if capacities or qps's differ.
+func (q *RateLimitedTimedQueue) SwapLimiter(newQPS float32) {
+	q.limiterLock.Lock()
+	defer q.limiterLock.Unlock()
+	if q.limiter.QPS() == newQPS {
+		return
+	}
+	var newLimiter flowcontrol.RateLimiter
+	if newQPS <= 0 {
+		newLimiter = flowcontrol.NewFakeNeverRateLimiter()
+	} else {
+		newLimiter = flowcontrol.NewTokenBucketRateLimiter(newQPS, evictionRateLimiterBurst)
+	}
+	// If we're currently waiting on limiter, we drain the new one - this is best effort approach.
+	for q.limiter.Saturation() > newLimiter.Saturation() {
```

Comment:
Well, in case of all limitters have Burst=1 it doesn't matter, but I still think that this method of saturation is bad.
Can you please at least add a TODO to consider changing it to be more numer-of-calls-oriented (as I described previously?).